.github/workflows/build-nightly.yml

```diff
@@ -79,8 +82,9 @@ jobs:
           path: setup\deploy\*.msi
 
   cleanup-job:
-    needs: get-latest-commit-timespan
-    if: needs.get-latest-commit-timespan.outputs.LATEST_COMMIT_TIMESPAN >= 86400
+    needs: build-nightly
+    # run only if the build-nightly job was skipped
+    if: always() && needs.build-nightly.result == 'skipped'
```

Comment:
Modified to avoid the duplication of the logic:
Run the `cleanup-job` only if the `build-nightly` job was skipped